File: Cpp/2576_find_the_maximum_number_of_marked_indices.cpp
// https://leetcode.cn/problems/find-the-array-concatenation-value/

class Solution {
public:
    int maxNumOfMarkedIndices(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int res = 0;
        int n = nums.size(), j = n/2;
        for (int i = 0; i < n/2; ++i) {
            int val = 2 * nums[i];
            while (j < n && val > nums[j]) {
                j += 1;
            }
            if (j < n) {
                res += 2;
                j += 1;
            }
            else {
                break;
            }
        }
        return res;
    }
};

File: Python3/2576_find_the_maximum_number_of_marked_indices.py
# https://leetcode.cn/problems/find-the-maximum-number-of-marked-indices/

class Solution:
    def maxNumOfMarkedIndices(self, nums):
        """
        Inputs:
            nums: list[int]
        Outputs:
            res: int
        """
        nums.sort()
        res = 0
        n, j = len(nums), len(nums)//2
        for i in range(n//2):
            val = 2 * nums[i]
            while j < n and val > nums[j]:
                j += 1
            if j < n:
                res += 2
                j += 1
            else:
                break
        return res